src/core/reactor.cc

```diff
@@ -2200,6 +2202,48 @@ void reactor::kill(pid_t pid, int sig) {
     ret.throw_if_error();
 }
 
+future<std::optional<struct group_details>> reactor::getgrnam(std::string_view name) {
+    syscall_result_extra<std::optional<struct group>> sr = co_await _thread_pool->submit<syscall_result_extra<std::optional<struct group>>>(
+        [name = sstring(name)] {
+            struct group grp;
+            struct group *result;
+            memset(&grp, 0, sizeof(struct group));
+            char buf[1024];
+            errno = 0;
+            int ret = ::getgrnam_r(name.c_str(), &grp, buf, sizeof(buf), &result);
+            return wrap_syscall(ret, result ? std::optional<struct group>(*result) : std::nullopt);
```

Comment:
I think there is a problem here, buf memory which holds the results (group info) is on the stack, and can go out of scope after the lambda.
The copty to group_details must be made here, instead of outside the lambda ?

Reply:
You can allocate it on the coroutine's stack, where the coroutine will save to its frame.